packages/itwinui-react/src/core/SearchBox/SearchBox.tsx

```diff
@@ -0,0 +1,398 @@
+    onToggle,
+    collapsedState,
+    isExpanded,
+    children,
+    inputProps,
+    openButtonProps,
+    closeButtonProps,
+    className,
+    ...rest
+  } = props;
+
+  const [inputId, setInputId] = React.useState(useId());
+  const inputRef = React.useRef<HTMLInputElement>(null);
+  const openButtonRef = React.useRef<HTMLButtonElement>(null);
+
+  const [localExpanded, setLocalExpanded] = React.useState(isExpanded);
+
+  const onClose = () => {
+    setLocalExpanded(false);
+    onToggle?.(false);
+    openButtonRef.current && openButtonRef.current.focus();
+  };
+
+  const onOpen = () => {
+    setLocalExpanded(true);
+    onToggle?.(true);
+    inputRef.current && inputRef.current.focus();
+  };
+
+  return (
+    <SearchBoxContext.Provider
+      value={{
+        size,
+        disabled,
+        onOpen,
+        onClose,
+        inputRef,
+        inputId,
+        setInputId,
+        openButtonRef,
+      }}
+    >
+      <InputFlexContainer
+        ref={ref}
+        className={cx({ 'iui-expandable-searchbox': expandable }, className)}
+        data-iui-size={size}
+        disabled={disabled}
+        role='searchbox'
+        data-iui-expanded={localExpanded}
+        {...rest}
+      >
+        {!expandable &&
+          (children ?? (
+            <>
+              <SearchBoxInput {...inputProps} />
+              <SearchBoxIcon />
+            </>
+          ))}
+        {expandable && (
+          <>
+            <div className='iui-searchbox-collapsed'>
+              {collapsedState ?? <SearchBoxOpenButton {...openButtonProps} />}
+            </div>
+            <div className='iui-searchbox-expanded'>
+              {children ?? (
+                <>
+                  <SearchBoxInput {...inputProps} />
+                  <SearchBoxCloseButton {...closeButtonProps} />
+                </>
+              )}
+            </div>
+          </>
+        )}
+      </InputFlexContainer>
+    </SearchBoxContext.Provider>
+  );
+}) as PolymorphicForwardRefComponent<
+  'div',
+  SearchBoxOwnProps & InputFlexContainerProps
+>;
+
+export type SearchBoxProps = PolymorphicComponentProps<
+  'div',
+  SearchBoxOwnProps & InputFlexContainerProps
+>;
+
+// ----------------------------------------------------------------------------
+export type SearchBoxIconProps = PolymorphicComponentProps<'span', IconProps>;
+
+const SearchBoxIcon = React.forwardRef((props, ref) => {
+  const { size, className, children, ...rest } = props;
+  const { size: sizeContext } = useSafeContext(SearchBoxContext);
+
+  return (
+    <Icon
+      aria-hidden
+      className={cx('iui-svg-icon', 'iui-search-icon', className)}
+      size={size ?? sizeContext}
+      ref={ref}
+      {...rest}
+    >
+      {children ?? <SvgSearch />}
+    </Icon>
+  );
+}) as PolymorphicForwardRefComponent<'span', IconProps>;
+
+// ----------------------------------------------------------------------------
+
+export type SearchBoxInputProps = {
+  label?: string;
```

Comment:
not necessary as user can just pass `aria-label` and we don't do anything else with `label`

Reply:
Removed label prop.
We used it for aria-label in case users won't add that on their own. I guess it's ok to trust them to add label or aria label to the searchbox.